cli/resource/process.go

```diff
@@ -28,22 +28,49 @@ type httpDirectory struct {
 }
 
 // Process processes the input.
-func Process(cli *cli.SensuCli, client *http.Client, input string, recurse bool, processor Processor) error {
+func Process(cli *cli.SensuCli, client *http.Client, inputs []string, recurse bool, processor Processor) error {
+	var resources []*types.Wrapper
+	for _, input := range inputs {
+		res, err := process(client, input, recurse)
+		if err != nil {
+			return err
+		}
+		resources = append(resources, res...)
+	}
+	if err := Validate(resources, cli.Config.Namespace()); err != nil {
+		return err
+	}
+	return processor.Process(cli.Client, resources)
+}
+
+func process(client *http.Client, input string, recurse bool) ([]*types.Wrapper, error) {
+	var resources []*types.Wrapper
 	urly, err := url.Parse(input)
 	if err != nil {
-		return err
+		return resources, err
 	}
+	var res []*types.Wrapper
 	if urly.Scheme == "" || len(urly.Scheme) == 1 {
 		// We are dealing with a file path
-		return ProcessFile(cli, input, recurse, processor)
+		res, err = ProcessFile(input, recurse)
+		if err != nil {
+			return resources, err
+		}
+	} else {
+		res, err = ProcessURL(client, urly, input, recurse)
+		if err != nil {
+			return resources, err
+		}
 	}
-	return ProcessURL(cli, client, urly, input, recurse, processor)
+	resources = append(resources, res...)
+	return resources, nil
 }
```

Comment:
Aren't `resources` and `res` redundant here? I don't understand why `res` is introduced. It looks to me like we could have used `resources` everywhere.

Reply:
`res` is meant to capture the resources returned by `ProcessFile` or `ProcessURL`.. I introduced the variable to limit the # of lines of code required, ex. I only have to call `resources = append(resources, res...)` once.